### DISCUSSION TITLE
How do I support keyboard events to enable Accessibility?

### DISCUSSION THREAD

**Discussion by TheEvilDev** [cytoscape/cytoscape.js#3125] [Q&A]
Hey Folks,

We are currently working with the cytoscape graph visualization component, and it's very compelling for our use case. One thing I can't find is documentation on how to make the visualization accessible.

Am I just missing it? Specifically, can we navigate nodes using keyboard events? Set focus targets, etc?

**Comment by maxkfranz** [comment]
See https://github.com/cytoscape/cytoscape.js/issues/3091#issuecomment-1387545550

> Accessibility is an important issue, and it's great that you're considering it for your app.
> 
> An app must be designed itself with the accessibility features you’d like to support in mind, taking into account the app’s use cases and the user’s mental model.
> 
> Let's take tabbing as a simple example:
> 
> 1. Do you allow tabbing through all the elements, or would users only want to tab through some of the elements?  Is it acceptable to force users to tab through potentially thousands of elements to get to the one they want?
> 2. In what order do you enable users to tab through elements?  By their geographic position?  By the graph topology?  Alphabetically?  By some app-specific semantics?
> 3. Do you force users to tab through elements based on where a Cytoscape instance just happens to exist in the DOM, or may the elements be tabbed through in a more suitable location?
> 
> These are but a few of the considerations you may make when designing your app w.r.t. accessibility. So, as you can see, it's important to tailor your solutions based on what is going to give a good experience for the users of your app specifically. Automated tools to assess accessibility, like Accessibility Insights for Web, are a good start, but they hardly paint the whole picture.
> 
> Generally, accessibility is something that needs to be built at the app level — especially because so many surrounding details depend on the design of the app itself. Cytoscape generally shouldn't, and can't, make those decisions for you.
> 
> In future, we may publish some accessibility reading materials re. Cytoscape. And since Cytoscape supports [extensions](https://js.cytoscape.org/#extensions), you're welcome to to publish reusable accessibility extensions for others who build similar apps to yours in future.
> 
> It's up to app authors like you to ensure that accessibility is properly addressed within the apps that they build. It's great that you consider this issue to be important, and we encourage you to continue to consider it as you design and develop your app.
